zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java

```diff
@@ -246,6 +246,13 @@ public void moveNote(String noteId,
 
     // update notebookrepo
     this.notebookRepo.move(noteId, notePath, newNotePath, subject);
+
+    // save note if note name is changed, because we need to update the note field in note json.
+    String oldNoteName = getNoteName(notePath);
+    String newNoteName = getNoteName(newNotePath);
+    if (!oldNoteName.equalsIgnoreCase(newNoteName)) {
```

Comment:
I think it would be good to use StringUtils.equalsIgnoreCase(..) to be null safe.

Reply:
Good point, Fixed